src/OrleansRuntime/Scheduler/ClosureWorkItem.cs

```diff
@@ -46,23 +47,113 @@ public override void Execute()
             continuation();
         }
 
-        public override WorkItemType ItemType { get { return WorkItemType.Closure; } }
+        public override WorkItemType ItemType => WorkItemType.Closure;
 
         #endregion
 
-        public override string ToString()
+        internal static string GetMethodName(Delegate action)
         {
-            var detailedName = string.Empty; 
-            if (nameGetter == null) // if NameGetter != null, base.ToString() will print its name.
+            var continuationMethodInfo = action.GetMethodInfo();
+            return string.Format(
```

Comment:
Why don't we set the formatted value in ctor and use a private getter to return the name? all the props are readonly and set during ctor, am I missing something?